src/main/java/graphql/execution/reactive/CompletionStageMappingOrderedPublisher.java

```diff
@@ -0,0 +1,36 @@
+package graphql.execution.reactive;
+
+import graphql.Internal;
+import org.jetbrains.annotations.NotNull;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+
+import java.util.concurrent.CompletionStage;
+import java.util.function.Function;
+
+/**
+ * A reactive Publisher that bridges over another Publisher of `D` and maps the results
+ * to type `U` via a CompletionStage, handling errors in that stage but keeps the results
+ * in order of downstream publishing. This means it must queue unfinished
+ * completion stages in memory in arrival order.
+ *
+ * @param <D> the downstream type
+ * @param <U> the upstream type to be mapped to
+ */
+@Internal
+public class CompletionStageMappingOrderedPublisher<D, U> extends CompletionStageMappingPublisher<D, U> {
```

Comment:
it extends `CompletionStageMappingPublisher` because they have a LOT in common - just the ordering on inflight objects is different

Reply:
Note also this has been split out of the publisher class